app/code/core/Mage/Adminhtml/Block/Customer/Edit/Tabs.php

```diff
@@ -80,13 +81,15 @@ protected function _beforeToHtml()
                 'url'       => $this->getUrl('*/*/wishlist', ['_current' => true]),
             ]);
 
-            /** @var Mage_Adminhtml_Block_Customer_Edit_Tab_Newsletter $block */
-            $block = $this->getLayout()->createBlock('adminhtml/customer_edit_tab_newsletter');
-            if (Mage::getSingleton('admin/session')->isAllowed('newsletter/subscriber')) {
-                $this->addTab('newsletter', [
-                    'label'     => Mage::helper('customer')->__('Newsletter'),
-                    'content'   => $block->initForm()->toHtml()
-                ]);
+            if (Mage::helper('core')->isModuleOutputEnabled('Mage_Newsletter')) {
+                /** @var Mage_Adminhtml_Block_Customer_Edit_Tab_Newsletter $block */
+                $block = $this->getLayout()->createBlock('adminhtml/customer_edit_tab_newsletter');
+                if (Mage::getSingleton('admin/session')->isAllowed('newsletter/subscriber')) {
+                    $this->addTab('newsletter', [
+                        'label'     => Mage::helper('customer')->__('Newsletter'),
+                        'content'   => $block->initForm()->toHtml()
+                    ]);
+                }
```

Comment:
I wonder if we can move this code out into the newsletter module and instead add the tab through layout xml 🤔. Maybe someone with more experience in this area can clarify if that's possible.

Reply:
+1 totally possible.

Reply:
> As suggested by @elidrissidev, we need to do a proper separation of concerns. `Mage_Customer` depends on the following modules:
> 
> https://github.com/OpenMage/magento-lts/blob/5318b8617612f07a2a10882c3c4eed262726eb3f/app/etc/modules/Mage_Customer.xml#L28-L32
> 
> All other modules here should be separated:
> 
> 1. `Mage_Sales`
> 2. `Mage_Catalog`
> 3. `Mage_Newsletter`
> 4. `Mage_Wishlist`
> 
> Besides the tabs, which can be easily added with the event `adminhtml_block_widget_tabs_html_before` (see PR #1358), the rendering of the pages need to be separated as well:
> 
> ![image](https://user-images.githubusercontent.com/1106470/224519685-629c12c2-a09f-4446-8387-db115fab36f6.png)
> 
> Related to PR #2455. It means also PR #2533. And down the rabbit hole. What should we do?

suggest we can first let the module sepreated into the repository.  later we can refactor one by one, I just afraid if we want a perfect solution and that may took too much time to completed.